components/brave_sync/time_limited_words_unittest.cc

```diff
@@ -0,0 +1,185 @@
+        TimeLimitedWords::GetWordsV1SunsetDay() - base::Days(1));
+    result = TimeLimitedWords::Validate(kValidSyncCode + std::string(" "),
+                                        &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kValid);
+    EXPECT_EQ(pure_words, kValidSyncCode);
+  }
+
+  {
+    // Invalid v1 sync code, prior to sunset date
+    auto time_override = OverrideWithTimeNow(
+        TimeLimitedWords::GetWordsV1SunsetDay() - base::Days(1));
+    result = TimeLimitedWords::Validate(kInvalidSyncCode, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kNotValidPureWords);
+    EXPECT_EQ(pure_words, "");
+  }
+
+  const base::Time anchorDayForWordsV2 =
+      QrCodeDataValidator::GetQRv1SunsetDay() + base::Days(20);
+  const std::string valid25thAnchoredWord =
+      TimeLimitedWords::GetWordByIndex(20);
+  const std::string valid25thAnchoredWords =
+      kValidSyncCode + std::string(" ") + valid25thAnchoredWord;
+
+  {
+    // Valid v2 sync code, after sunset date, around anchored day
+    auto time_override = OverrideWithTimeNow(anchorDayForWordsV2);
+    result = TimeLimitedWords::Validate(valid25thAnchoredWords, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kValid);
+    EXPECT_EQ(pure_words, kValidSyncCode);
+  }
+
+  {
+    // Valid v2 sync code, after sunset date, expired
+    const std::string valid25thExpiredWord =
+        TimeLimitedWords::GetWordByIndex(15);
+    const std::string valid25thExpiredWords =
+        kValidSyncCode + std::string(" ") + valid25thExpiredWord;
+
+    auto time_override = OverrideWithTimeNow(anchorDayForWordsV2);
+    result = TimeLimitedWords::Validate(valid25thExpiredWords, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kExpired);
+    EXPECT_EQ(pure_words, "");
+  }
+
+  {
+    // Valid v2 sync code, after sunset date, valid for too long
+    const std::string valid25thValidTooLongWord =
+        TimeLimitedWords::GetWordByIndex(25);
+    const std::string valid25thValidTooLongWords =
+        kValidSyncCode + std::string(" ") + valid25thValidTooLongWord;
+
+    auto time_override = OverrideWithTimeNow(anchorDayForWordsV2);
+    result =
+        TimeLimitedWords::Validate(valid25thValidTooLongWords, &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kValidForTooLong);
+    EXPECT_EQ(pure_words, "");
+  }
+
+  {
+    // Wrong words number
+    auto time_override = OverrideWithTimeNow(anchorDayForWordsV2);
+    result = TimeLimitedWords::Validate("abandon ability", &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kWrongWordsNumber);
+    EXPECT_EQ(pure_words, "");
+
+    result = TimeLimitedWords::Validate(
+        valid25thAnchoredWords + " abandon ability", &pure_words);
+    EXPECT_EQ(result, WordsValidationResult::kWrongWordsNumber);
+    EXPECT_EQ(pure_words, "");
+  }
+
+  {
+    // Valid v2 sync code, after sunset date, day modulo 2048 which is
+    // "2027-08-11 00:00:00.000 UTC"
```

Comment:
This test case is correct, but I would add a note to say that it's a limitation of this simplistic expiry mechanism. Maybe something like:
```
// Note: While this date is way too far into the future, the codes repeat after a few years and so this
// becomes valid again, an unfortunate limitation of this scheme.
```

Reply:
thanks, this is fixed at https://github.com/brave/brave-core/pull/12300/commits/a780c86e39a3a7e2716e91c82d666607d0ed0709